main.py

```diff
@@ -33,30 +44,32 @@ def setupSpotifyObject(username, scope, clientID, clientSecret, redirectURI):
     token = util.prompt_for_user_token(username, scope, clientID, clientSecret, redirectURI)
     return spotipy.Spotify(auth=token)
 
-def main(username, scope, clientID, clientSecret, redirectURI, path):    
+def main(username, scope, clientID, clientSecret, redirectURI, path):
     spotify = setupSpotifyObject(username, scope, clientID, clientSecret, redirectURI)
 
+    restartSpotify(path)
+
     while True:
-        
+
         try:
             current_track = spotify.current_user_playing_track()
-        except:
+        except Exception:

```

Comment:
Doesn't the API throw a less broad exception when tokens expire?

Reply:
Does it throw a `spotipy.exceptions.SpotifyException`? I'll specify that, but I couldn't ever get that exception to be thrown even when I removed the application

Reply:
I just checked the code and I believe it does raise a `spotipy.exceptions.SpotifyException`, which subclasses `Exception`, so either way would probably be fine. I'll go ahead and just use the more specific one